### PR TITLE
KHWE-13-Implement-share-view

### PR DESCRIPTION
**Pull request by gamesminer**
screencapture-localhost-8080-2018-10-31-22_18_11

### CODE REVIEW

**Review by OlegShynkarenko — APPROVED** [review]

**Comment by ihor4568** [comment]
Could you please use default card look (I mean default font-weight for caption and text and default margins). Like here

image

Also please put this card in the center of the page

File: src/components/Main/Main.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      </div>
+    </div>
+  </aside>
+  <div class="main__wrapper mdc-drawer-app-content">
+    <header class="main__header mdc-top-app-bar app-bar"></header>
+    <main class="main__section main-content">
+      <div class="main__content-mount mdc-top-app-bar--fixed-adjust">
+        <h5 class="mdc-typography--headline3">Albums</h5>
+      </div>
+      <div class="main__share-view"></div>
```
Comment (ihor4568) [inline src/components/Main/Main.html]: there should no be specific mount point for share view. It should be mounted into common mount point where albums, artists and about view mount

Comment (ihor4568) [reply src/components/Main/Main.html]: probably inside `main__content-mount`

File: src/components/SharingSong/SharingSong.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div class="share-view">
```
Comment (ihor4568) [inline src/components/SharingSong/SharingSong.html]: style class name should be the same as component name just in kebab-case

File: src/components/SharingSong/SharingSong.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { MDCRipple } from "@material/ripple";
+import SharingSongTemplate from "./SharingSong.html";
```
Comment (ihor4568) [inline src/components/SharingSong/SharingSong.js]: this should start from lover case letter - it is function

File: src/components/SharingSong/SharingSong.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export class SharingSongComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  initMaterialCard() {
+    const cardRipple = new MDCRipple(
+      this.mountPoint.querySelector(".share-view__card-riple")
```
Comment (ihor4568) [inline src/components/SharingSong/SharingSong.js]: this should be in queryselectors method

File: src/components/SharingSong/SharingSong.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { MDCRipple } from "@material/ripple";
+import SharingSongTemplate from "./SharingSong.html";
+
+export class SharingSongComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  initMaterialCard() {
```
Comment (ihor4568) [inline src/components/SharingSong/SharingSong.js]: this should be called initMaterial

**Comment by ihor4568** [comment]
The font weight, font sizes and colors of text and paddings/margins should be the same as in default material card

image

compare to yours

image

File: src/components/ShareView/ShareView.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  albumName: "Loud",
+  artistName: "by Rihanna",
+  imageSource: "https://image.ibb.co/k4Qc8L/rihanna-loud.jpg"
+};
+
+export class ShareViewComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+    this.song = SONG;
```
Comment (ghost) [inline src/components/ShareView/ShareView.js]: pass `SONG` into the props or simply access as is

File: package.json [hunk 7]
```diff
@@ -49,6 +50,7 @@
   },
   "lint-staged": {
     "*.js": [
+      "prettier --write",
```
Comment (ghost) [inline package.json]: why we need it here if it is listed below

File: src/components/DotsMenu/DotsMenu.js [hunk 8]
```diff
@@ -43,6 +44,7 @@ export class DotsMenuComponent {
     this.querySelectors();
     this.initMaterial();
     this.addEventListeners();
+    // console.dir(this.shareDot[this.shareDot]);
```
Comment (ihor4568) [inline src/components/DotsMenu/DotsMenu.js]: please remove this comment

**Review by ihor4568 — APPROVED** [review]

**Pull request merged**